pkg/operator/staticpod/installerpod/cmd.go

```diff
@@ -398,6 +400,56 @@ func (o *InstallOptions) substituteSecret(obj *corev1.Secret) *corev1.Secret {
 	return ret
 }
 
+func (o *InstallOptions) checkForHigherRevision(ctx context.Context) (error, bool) {
+	higherRevisionFound := false
+	targetRevision, err := strconv.Atoi(o.Revision)
+	if err != nil {
+		return fmt.Errorf("unable to convert revision into an integer: %v", err), false
+	}
+
+	// Make sure at most one installer carries this check
+	installerLockFile := path.Join(o.ResourceDir, o.PodConfigMapNamePrefix+".lock")
+	installerLock := flock.New(installerLockFile)
+	klog.Infof("acquiring an exclusive lock on a %s", installerLockFile)
+	if err := installerLock.Lock(ctx); err != nil {
+		return fmt.Errorf("failed to acquire an exclusive lock on %s, due to %v", installerLockFile, err), false
+	}
+	defer installerLock.Unlock()
+
+	if err := filepath.Walk(o.ResourceDir, func(filePath string, info os.FileInfo, err error) error {
```

Comment:
how does a lock around the directory walker protect concurrent directory+file creation? Seems to be racy.

Reply:
You need to walk through the directory before you can create it. The lock is release after you know whether you can create the directory or not. So when two installers are running in parallel, only one of them is checking for the revision at the given time.

Unless you get to a situation where installer 6 and 7 are running at the same time. Installer 6 takes the precedence when checking for the revision. At which point it gets green. After which the revision 7 installer checks the revision and gets green as well since no matter when the installer 6 creates the directory, installer 7 will always have higher revision. At which point revision 6 installer might take a longer and actually rewrite the revision 7 resources.